lib/batch.js

```diff
@@ -44,17 +49,21 @@ function batch(request, map) {
           return undefined;
         default:
           logger.error(prop + '.' + res.statusCode, res.result);
-          throw AppError.from(res.result);
+          return AppError.from(res.result);
       }
     });
   });
-  return P.props(result).then(function(result) {
+  P.props(result).then(function(result) {
     Object.keys(result).forEach(function(key) {
       if (result[key] === undefined) {
         delete result[key];
+      } else if (result[key].isBoom) {
+        return next(result[key]);
       }
     });
-    return result;
+    const shouldCache = config.serverCache.enabledEmailAddresses.test(email);
```

Comment:
So if memory is the default engine, let's just use the regex for full OFF / ON switch of the feature. We should be already able to do it without any changes. 👍